plugins/arElasticSearchPlugin/lib/arElasticSearchMultiIndexWrapper.class.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * arElasticSearchMultiIndexWrapper facilitates handling ElasticSearch indices
+ * and has methods that match signatures of pre ES 6.x methods that used a
+ * single index with multiple types instead of multiple index with a single
+ * type or no type.
+ */
+class arElasticSearchMultiIndexWrapper
+{
+    protected $indices;
+
+    public function __construct()
+    {
+        $this->indices = [];
+    }
+
+    public function addIndex($name, Elastica\Index $index)
+    {
+        $this->indices[$name] = $index;
+    }
+
+    public function delete()
+    {
+        foreach ($this->indices as $index) {
+            $index->delete();
+        }
+    }
```

Comment:
Given all the feedback above, this would be the only function without a `$name` parameter in the wrapper (apart from the constructor). And, when we revisit the `exclude-types` option I think we'll add that parameter.

It makes a lot of sense to me that all methods from this class require a name parameter now that it's a multi index wrapper instead of an index decorator.

It would be even clearer if `$index` was called `$indicesWrapper` in `arElasticSearchPlugin`, but preventing issues in custom code is a stronger point than clarity.

Just a note.